### PR TITLE
Integrate API: edit exam's description

### PR DESCRIPTION
**Pull request by Wally5077**
resolve: #100 

image

### CODE REVIEW

**Review by nyngwang — APPROVED** [review]
LGTM.

File: src/app/exam/questions/ExamQuestions.js [hunk 1]
```diff
@@ -100,6 +100,18 @@ const ExamQuestions = () => {
         return examService.addExamQuestion(question).then(fetchExam);
     };
 
+    const onExamDescriptionSaved = (description) => {
+        if (currentExam.description !== description) {
+            examService.updateExam(examId, {
+                examId: examId,
```
Comment (Johnny850807) [inline src/app/exam/questions/ExamQuestions.js:106]: `examId: examId` --> `examId`

File: src/app/commons/MarkdownEditor.js [hunk 2]
```diff
@@ -18,48 +17,32 @@ function Tabs({textareaVal, setTextareaVal}) {
...
         </div>
     );
 }
 
-const MarkdownEditor = ({style}) => {
+const MarkdownEditor = ({text, onSaved, style}) => {
     // TODO: problemService.getProblemDescription
     const [editingState, setEditingState] = useState(false);
-    const [textareaVal, setTextareaVal] = useState('Press Edit Description to start writing the description. Styling with Markdown is supported.\n');
+    const [textareaVal, setTextareaVal] = useState(text ? text : 'Press Edit Description to start writing the description. Styling with Markdown is supported.\n');
```
Comment (Johnny850807) [inline src/app/commons/MarkdownEditor.js:45]: `text || 'Press Edit Description to start writing the description. Styling with Markdown is supported.\n'`

File: src/app/commons/MarkdownEditor.js [hunk 3]
```diff
@@ -18,48 +17,32 @@ function Tabs({textareaVal, setTextareaVal}) {
...
+                <div className={(toggleState === TAB_PREVIEW ? "active-markdown" : "hide")}>
+                    <ReactMarkdown>{textareaVal}</ReactMarkdown>
                 </div>
             </div>
         </div>
     );
 }
 
-const MarkdownEditor = ({style}) => {
+const MarkdownEditor = ({text, onSaved, style}) => {
```
Comment (Johnny850807) [inline src/app/commons/MarkdownEditor.js:42]: Don't do `onSaved`
do `onTextChanged`

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done

**Review by timt1028 — APPROVED** [review]

File: src/app/commons/MarkdownEditor.js (hunk 2 above)
Comment (Wally5077) [reply src/app/commons/MarkdownEditor.js:45]: I gotta say this statement is truly interesting .

Comment (Johnny850807) [reply src/app/commons/MarkdownEditor.js:45]: Does this work?

Comment (Wally5077) [reply src/app/commons/MarkdownEditor.js]: It does work.

File: src/app/commons/MarkdownEditor.js [hunk 4]
```diff
@@ -118,8 +96,7 @@ const MarkdownEditor = ({style}) => {
                     fontSize={15}
                     borderRadius={20}
                     marginTop={15} marginRight={4}
-                    onClick={e => handleDescription(e)}
-                />
+                    onClick={e => handleDescription(e)}/>
```
Comment (Johnny850807) [inline src/app/commons/MarkdownEditor.js:99]: move the button out of the `MarkdownEditor`

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done

File: src/app/commons/MarkdownEditor.js (hunk 3 above)
Comment (Johnny850807) [reply src/app/commons/MarkdownEditor.js]: image

**Review by Johnny850807 — APPROVED** [review]
LGTM

**Pull request merged**